src/cart/cart.ts

```diff
@@ -28,3 +34,19 @@ export const calculateProductLimit = (productId: string, currentLimit: number):
     Object.values(carts).reduce((acc: number, cart: Cart) =>
         cart[productId] ? acc - cart[productId] : acc, currentLimit);
 
+
+export const updateProductInCart = async (userId: string, productId: string, amount: number): Promise<Product> => {
```

Comment:
all of this is ok, but when you throw gql errors to the user you should use apollo's errors